mycroft/util/time.py

```diff
@@ -0,0 +1,95 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2018 Mycroft AI Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from datetime import datetime
+from dateutil.tz import gettz, tzlocal
+
+
+def default_timezone():
+    """ Get the default timezone
+
+    Based on user location settings location.timezone.code or
+    the default system value if no setting exists.
+
+    Returns:
+        (datetime.tzinfo): Definition of the default timezone
+    """
+    try:
+        # Obtain from user's configurated settings
+        #   location.timezone.code (e.g. "America/Chicago")
+        #   location.timezone.name (e.g. "Central Standard Time")
+        #   location.timezone.offset (e.g. -21600000)
+        from mycroft.configuration import Configuration
+        config = Configuration.get()
+        code = config["location"]["timezone"]["code"]
+
+        return gettz(code)
+    except Exception:
+        # Just go with system default timezone
+        return tzlocal()
+
+
+def now_utc():
+    """ Retrieve the current time in UTC
+
+    Returns:
+        (datetime): The current time in Universal Time, aka GMT
+    """
+    return datetime.utcnow()
+
+
+def now_local(tz=None):
+    """ Retrieve the current time
+
+    Args:
+        tz (datetime.tzinfo, optional): Timezone, default to user's settings
+
+    Returns:
+        (datetime): The current time
+    """
+    if not tz:
+        tz = default_timezone()
+    return datetime.now(tz)
+
+
+def to_utc(dt):
+    """ Convert a datetime with timezone info to a UTC datetime
+
+    Args:
+        dt (datetime): A datetime (presumably in some local zone)
+    Returns:
+        (datetime): time converted to UTC
+    """
+    tzUTC = gettz("UTC")
+    if dt.tzinfo:
+        return dt.astimezone(tzUTC)
+    else:
+        return dt.replace(tzinfo=gettz("UTC")).astimezone(tzUTC)
```

Comment:
Isn't it enough to just do the 

`dt.replace(tzinfo=gettz("UTC"))`

The `astimezone()` would convert from UTC to UTC right?

Reply:
I understand, I just thought the `astimezone` was redundant in this case (I only did a quick test so I might be wrong) but I trust you that it's necessary.

Reply:
Yeah, we could special-case to skip calling astimezone() in the UTC -> UTC case, but astimezone() already has the code to handle it.